lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -54,26 +56,73 @@ internal sealed class FileSystemInputPartition<T> : IInputPartition<T>, IDisposa
             _id = id;
             _fileSystem = fileSystem;
             _fileSerializer = fileSerializer;
-            _filePaths = filePaths;
             _tempFileCreator = tempFileCreator;
-            _isInitialized = false;
-            _copyToLocal = copyToLocal;
+            _filePaths = filePaths;
+
+            if (!copyToLocal)
+            {
+                // Implies that the files are already local.
```

Comment:
copyToLocal means download from remote to local before reading the data. 
!copyToLocal means read data from file (no matter files are on local or remote). It doesn't mean the file is already in local. 

In our use case, the  files are always remote. 

For local test, the files can be at local, then copyToLocal is redundant, but still work.